sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -2584,6 +2584,86 @@ gather_impl(AccessorTy acc, simd<uint32_t, N> offsets, uint32_t glob_offset,
   }
 }
 
+/// SLM gather implementation.
+/// Supported platforms: DG2, PVC
+/// VISA instruction: lsc_load.slm
+///
+/// Collects elements located at slm and returns them
+/// as a single \ref simd object.
+///
+/// @tparam T is element type.
+/// @tparam NElts is the number of elements to load per address.
+/// @tparam DS is the data size.
+/// @tparam N is the number of channels (platform dependent).
+/// @param offsets is the zero-based offsets for SLM buffer in bytes.
+/// @param pred is predicates.
+/// @return is a vector of type T and size N * NElts
+///
+
+template <typename T, int NElts, lsc_data_size DS, int N>
+__ESIMD_API __ESIMD_NS::simd<T, N * NElts>
+slm_gather_impl(__ESIMD_NS::simd<uint32_t, N> offsets,
+                __ESIMD_NS::simd_mask<N> pred) {
```

Comment:
This function can be removed. Please keep only the version with `pass_thru` operand. When don't have such operand, simply pass some uninitialized pass_thru to this gather_impl function.
The code must be the same/identical.

Reply:
That is what I talking about. Let's use only __esimd_lsc_load_merge_slm in all cases.

Reply:
```C++
  /// Default constructor. Values of the constructed object's elements are
  /// undefined.
  simd_obj_impl() = default;
```